plugins/workspace-plugin/src/utils.ts

```diff
@@ -0,0 +1,94 @@
+/*********************************************************************
+ * Copyright (c) 2020 Red Hat, Inc.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ **********************************************************************/
+
+import { che as cheApi } from '@eclipse-che/api';
+import * as che from '@eclipse-che/plugin';
+import * as theia from '@theia/plugin';
+import { execute } from './exec';
+
+export async function getEditorContainerName(): Promise<string | undefined> {
+    const TYPE: string = 'type';
+    const EDITOR_SERVER_TYPE: string = 'ide';
+    const containers = await getContainerList();
+    for (const container of containers) {
+        const servers = container.servers || {};
+        for (const serverName in servers) {
+            if (!servers.hasOwnProperty(serverName)) {
+                continue;
+            }
+            const attrs = servers[serverName].attributes || {};
+            for (const attrName in attrs) {
+                if (attrName === TYPE && attrs[attrName] === EDITOR_SERVER_TYPE) {
+                    return container.name;
+                }
+            }
+        }
+    }
+}
+
+export async function getContainerList(): Promise<WorkspaceContainer[]> {
+    const containers: WorkspaceContainer[] = [];
+    try {
+        const workspace = await che.workspace.getCurrentWorkspace();
+
+        if (workspace.runtime && workspace.runtime.machines) {
+            const machines = workspace.runtime.machines;
+            for (const machineName in machines) {
+                if (!machines.hasOwnProperty(machineName)) {
+                    continue;
+                }
+                const container: WorkspaceContainer = { name: machineName, ...machines[machineName] };
+                containers.push(container);
+            }
+        }
+    } catch (e) {
+        throw new Error('Unable to get list workspace containers. Cause: ' + e);
+    }
+    return containers;
+}
+
+export async function execInTerminal(directory: string, locationURI: string, ...args: string[]): Promise<void> {
+    const editorContainerName = await getEditorContainerName() as string;
+    try {
+        const shellArgs: string[] = ['-c', 'git'].concat(args);
```

Comment:
Should `git` come as arg?
I mean it's `utils` and `execInTerminal` function, so why `git` is hard coded here?

Reply:
Indeed. I think it should be possible to run `git` instead of `sh` in the following code.

Reply:
which comes as arg and is not hard coded
 :wink:

btw, the function executes a command in editor's container, so it should be defined in function name or the function should have the corresponding docs/comment or a container name should come as arg as well